src/components/hardware/HardwareWalletTheme.tsx

```diff
@@ -1,38 +1,81 @@
-import { type DefaultTheme, ThemeProvider } from "styled-components";
-import { type PropsWithChildren, useMemo } from "react";
+import { ThemeProvider, type DefaultTheme } from "styled-components";
+import { useEffect, type PropsWithChildren } from "react";
 import { useHardwareApi } from "~wallets/hooks";
+import { useTheme } from "~utils/theme";
+import { Provider } from "@arconnect/components";
+import { useTheme as useStyledComponentsTheme } from "styled-components";
+import { MotionGlobalConfig } from "framer-motion";
 
-export default function HardwareWalletTheme({
-  children
-}: PropsWithChildren<{}>) {
-  // hardware api type
+const ARCONNECT_THEME_BACKGROUND_COLOR = "ARCONNECT_THEME_BACKGROUND_COLOR";
+
+/**
+ * Modify the theme if the active wallet is a hardware wallet. We transform the
+ * default accent color to match the hardware wallet's accent.
+ */
+function hardwareThemeModifier(theme: DefaultTheme): DefaultTheme {
+  return {
+    ...theme,
+    theme: "154, 184, 255",
+    primary: "#9AB8FF",
+    primaryBtnHover: "#6F93E1"
+  };
+}
+
+function noThemeModifier(theme: DefaultTheme): DefaultTheme {
+  return theme;
+}
+
+export function ArConnectThemeProvider({ children }: PropsWithChildren<{}>) {
   const hardwareApi = useHardwareApi();
+  const theme = useTheme();
+  const themeModifier = hardwareApi ? hardwareThemeModifier : noThemeModifier;
+
+  useEffect(() => {
+    const reducedMotionPreference = window.matchMedia(
```

Comment:
Added this here. It could eventually be added in settings too, just like the system theme can be overridden.